app/src/main/kotlin/com/softteco/template/data/di/DataStoreModule.kt

```diff
@@ -24,49 +26,60 @@ import kotlinx.coroutines.SupervisorJob
+private const val USER_PROFILE_ENCRYPTED = "user_profile_encrypted_data"
+private const val AUTH_TOKEN_ENCRYPTED = "auth_token_encrypted"
 
 @Module
 @InstallIn(SingletonComponent::class)
 object DataStoreModule {
     @Provides
     @Singleton
-    fun providePreferencesDataStore(@ApplicationContext appContext: Context): DataStore<Preferences> {
-        return PreferenceDataStoreFactory.create(
+    @Named("authToken")
+    fun providePreferencesDataStore(@ApplicationContext appContext: Context): DataStore<AuthTokenDto> {
+        return DataStoreFactory.create(
             corruptionHandler = ReplaceFileCorruptionHandler(
-                produceNewData = { emptyPreferences() }
+                produceNewData = { AuthTokenDto("") }
             ),
-            migrations = listOf(SharedPreferencesMigration(appContext, USER_PREFERENCES)),
             scope = CoroutineScope(Dispatchers.IO + SupervisorJob()),
-            produceFile = { appContext.preferencesDataStoreFile(USER_PREFERENCES) }
+            produceFile = { appContext.dataStoreFile(AUTH_TOKEN_ENCRYPTED) },
+            serializer = AuthTokenSerializer(CryptoManager())
         )
     }
 
     @Provides
     @Singleton
-    fun provideEncryptedDataStore(@ApplicationContext appContext: Context): DataStore<CreateUserDto> {
+    fun provideUserProfileEncryptedDataStore(@ApplicationContext appContext: Context): DataStore<ProfileDto> {
         return DataStoreFactory.create(
             corruptionHandler = ReplaceFileCorruptionHandler(
-                produceNewData = { CreateUserDto("", "", "") }
+                produceNewData = { ProfileDto(0, "", "", "", "") }
             ),
             scope = CoroutineScope(Dispatchers.IO + SupervisorJob()),
-            produceFile = { appContext.dataStoreFile(USER_ENCRYPTED) },
-            serializer = UserSettingsSerializer(CryptoManager())
+            produceFile = { appContext.dataStoreFile(USER_PROFILE_ENCRYPTED) },
+            serializer = UserProfileSerializer(CryptoManager())
         )
     }
 
     @Provides
     @Singleton
-    @Named("themeMode")
-    fun provideThemeModeDataStore(@ApplicationContext appContext: Context): DataStore<Preferences> {
+    fun provideDataStore(
+        @ApplicationContext appContext: Context,
+        preferencesFileName: String
+    ): DataStore<Preferences> {
         return PreferenceDataStoreFactory.create(
             corruptionHandler = ReplaceFileCorruptionHandler(
                 produceNewData = { emptyPreferences() }
             ),
+            migrations = listOf(SharedPreferencesMigration(appContext, preferencesFileName)),
             scope = CoroutineScope(Dispatchers.IO + SupervisorJob()),
-            produceFile = { appContext.preferencesDataStoreFile(APP_THEME_PREFERENCES) }
+            produceFile = { appContext.preferencesDataStoreFile(preferencesFileName) }
         )
     }
+
+    @Provides
+    @Singleton
+    @Named("themeMode")
```

Comment:
This function could be removed, we can use one not encrypted data store. Which you have above. 

```
    @Provides
    @Singleton
    fun provideDataStore(
        @ApplicationContext appContext: Context,
    ): DataStore<Preferences> {
        return PreferenceDataStoreFactory.create(
            corruptionHandler = ReplaceFileCorruptionHandler(
                produceNewData = { emptyPreferences() }
            ),
            migrations = listOf(SharedPreferencesMigration(appContext, APP_THEME_PREFERENCES)),
            scope = CoroutineScope(Dispatchers.IO + SupervisorJob()),
            produceFile = { appContext.preferencesDataStoreFile(APP_THEME_PREFERENCES) }
        )
    }
```